Source/Core/Event.js

```diff
@@ -46,7 +47,7 @@ Object.defineProperties(Event.prototype, {
  * An optional scope can be provided to serve as the <code>this</code> pointer
  * in which the function will execute.
  *
- * @param {Function} listener The function to be executed when the event is raised.
+ * @param {Listener} listener The function to be executed when the event is raised.
```

Comment:
This ends up creating a non-existent `Listener` entry in the doc:

![image](https://user-images.githubusercontent.com/1495196/83272336-04c04880-a199-11ea-8d7e-1dad51694a16.png)

I started to dig into template/generics a little on the JSDoc and I agree they don't appear to work at first glance. I'll see what I can find.  I want to be careful about trying to do too much too fast since we want to use generics in more places (like Property) but I don't want to get too crazy with workaround either.

Reply:
I totally forgot to link you to https://github.com/englercj/tsd-jsdoc/issues/134 -- it looks like `@template` is flat-out broken, but also [there isn't consensus about how to provide a default value for a type parameter](https://github.com/microsoft/TypeScript/issues/29401).  Sadly that probably means the ugly hack in the gulpfile will be required for the foreseeable future if you're going to use generics.